src/handlers.rs

```diff
@@ -254,63 +67,67 @@ pub extern "C" fn handle_code_fetch(
 
-  debug!(
-        "code_fetch. module_name = {} module_specifier = {} containing_file = {} filename = {}",
-        module_name, module_specifier, containing_file, filename
-    );
+  assert!(deno.dir.root.join("gen") == deno.dir.gen, "Sanity check");
 
-  if is_remote(&module_name) {
-    unimplemented!();
-  } else if module_name.starts_with(ASSET_PREFIX) {
-    assert!(false, "Asset resolution should be done in JS, not Rust.");
-  } else {
-    assert!(
-      module_name == filename,
-      "if a module isn't remote, it should have the same filename"
-    );
-    let result = File::open(&filename);
-    if result.is_err() {
-      let err = result.unwrap_err();
-      let errmsg = format!("{} {}", err, filename);
-      reply_error(d, cmd_id, &errmsg);
-      return;
-    }
-    let mut f = result.unwrap();
-    let result = f.read_to_string(&mut source_code);
-    if result.is_err() {
-      let err = result.unwrap_err();
-      let errmsg = format!("{} {}", err, filename);
+  let result = deno
+    .dir
+    .code_fetch(module_specifier, containing_file)
+    .map_err(|err| {
+      let errmsg = format!("{}", err);
       reply_error(d, cmd_id, &errmsg);
-      return;
-    }
+    });
+  if result.is_err() {
+    return;
   }
-
-  let output_code = String::new(); //load_output_code_cache(filename, source_code);
-
-  reply_code_fetch(
-    d,
-    cmd_id,
-    &module_name,
-    &filename,
-    &source_code,
-    &output_code,
-  )
+  let out = result.unwrap();
+  // reply_code_fetch
+  let mut builder = flatbuffers::FlatBufferBuilder::new();
+  let mut msg_args = msg::CodeFetchResArgs {
+    module_name: builder.create_string(&out.module_name),
+    filename: builder.create_string(&out.filename),
+    source_code: builder.create_string(&out.source_code),
+    ..Default::default()
+  };
+  match out.maybe_output_code {
+    Some(ref output_code) => {
+      msg_args.output_code = builder.create_string(output_code);
+    }
+    _ => (),
+  };
+  let msg = msg::CreateCodeFetchRes(&mut builder, &msg_args);
+  builder.finish(msg);
+  let args = msg::BaseArgs {
+    cmdId: cmd_id,
+    msg: Some(msg.union()),
+    msg_type: msg::Any::CodeFetchRes,
+    ..Default::default()
+  };
+  set_response_base(d, &mut builder, &args)
 }
 
-fn is_remote(_module_name: &String) -> bool {
-  false
+// https://github.com/ry/deno/blob/golang/os.go#L156-L169
+#[no_mangle]
+pub extern "C" fn handle_code_cache(
+  d: *const DenoC,
+  cmd_id: u32,
+  filename_: *const c_char,
+  source_code_: *const c_char,
+  output_code_: *const c_char,
+) {
+  let deno = from_c(d);
+  let filename = str_from_ptr!(filename_);
+  let source_code = str_from_ptr!(source_code_);
+  let output_code = str_from_ptr!(output_code_);
+  let result = deno.dir.code_cache(filename, source_code, output_code);
+  if result.is_err() {
+    let err = result.unwrap_err();
+    let errmsg = format!("{}", err);
+    reply_error(d, cmd_id, &errmsg);
+  }
+  // null response indicates success.
```

Comment:
Implicit null response is a foot gun. Probably better to explicitly say reply_null() if a handler succeeds without a return value.

Reply:
Agree. I think this whole interface is going to change dramatically soon. This is just a first pass. I'll leave it as is now.